src/mscorlib/src/System/Array.cs

```diff
@@ -976,7 +976,7 @@ public Object Clone()
             if (array==null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
             Contract.EndContractBlock();
-            return BinarySearch<T>(array, 0, array.Length, value, null);
+            return BinarySearch<T>(array, 0, array.Length, value, (IComparer<T>)null);
```

Comment:
The fact that you had to add such casts to existing code is a red flag. This means that existing code that makes use of the existing API will no longer compile without similar fixes. There's a reason why API additions are required to be reviewed first. Sure, they could probably review the PR as is but you may find that you have wasted your time implementing the wrong thing.